bib/cmd/bootc-image-builder/main.go

```diff
@@ -178,6 +184,13 @@ func manifestFromCobra(cmd *cobra.Command, args []string) ([]byte, *mTLSConfig,
 	tlsVerify, _ := cmd.Flags().GetBool("tls-verify")
 	localStorage, _ := cmd.Flags().GetBool("local")
 	rootFs, _ := cmd.Flags().GetString("rootfs")
+	pull, _ := cmd.Flags().GetString("pull")
+
+	if localStorage {
+		// don't error out since --local and
+		// --pull=never are functionally equivalent
+		pull = string(never)
```

Comment:
Maybe could be a bit stricter with the error checking here about the inputs we got from pull=<input>? To ensure that only valid values are input? And maybe error if pull and localStorage are set to conflicting values?

As a strawman for the validation, maybe something like (but there are many ways to do this, another alternative might be https://pkg.go.dev/github.com/thediveo/enumflag#section-readme to make sure that the inputs are checked on the cobra level).
```go
diff --git a/bib/cmd/bootc-image-builder/main.go b/bib/cmd/bootc-image-builder/main.go
index 94841f9..d6982e9 100644
--- a/bib/cmd/bootc-image-builder/main.go
+++ b/bib/cmd/bootc-image-builder/main.go
@@ -32,6 +32,15 @@ import (
 
 type pullPolicy string
 
+func newPullPolicy(s string) (pullPolicy, error) {
+       switch pullPolicy(s) {
+       case pullPolicyAlways, pullPolicyNever, pullPolicyMissing:
+               return pullPolicy(s), nil
+       default:
+               return "", fmt.Errorf("unknown pull policy %q", s)
+       }
+}
+
 const (
        // As a baseline heuristic we double the size of
        // the input container to support in-place updates.
@@ -39,9 +48,9 @@ const (
        // future.
        containerSizeToDiskSizeMultiplier = 2
 
-       always  pullPolicy = "always"
-       never   pullPolicy = "never"
-       missing pullPolicy = "missing"
+       pullPolicyAlways  pullPolicy = "always"
+       pullPolicyNever   pullPolicy = "never"
+       pullPolicyMissing pullPolicy = "missing"
 )
 
 // all possible locations for the bib's distro definitions
@@ -184,12 +193,16 @@ func manifestFromCobra(cmd *cobra.Command, args []string) ([]byte, *mTLSConfig,
        tlsVerify, _ := cmd.Flags().GetBool("tls-verify")
        localStorage, _ := cmd.Flags().GetBool("local")
        rootFs, _ := cmd.Flags().GetString("rootfs")
-       pull, _ := cmd.Flags().GetString("pull")
+       pullFlag, _ := cmd.Flags().GetString("pull")
 
        if localStorage {
                // don't error out since --local and
                // --pull=never are functionally equivalent
-               pull = string(never)
+               pullFlag = string(pullPolicyNever)
+       }
+       imagePullPolicy, err := newPullPolicy(pullFlag)
+       if err != nil {
+               return nil, nil, err
        }
 
        if targetArch != "" && arch.FromString(targetArch) != arch.Current() {
@@ -248,11 +261,10 @@ func manifestFromCobra(cmd *cobra.Command, args []string) ([]byte, *mTLSConfig,
                return arch.FromString(a) == cntArch
        }
 
-       imagePullPolicy := pullPolicy(pull)
        // If --pull=always or if --pull=missing and the container image is missing,
        // pull the container into local storage. This behaviour is more in line
        // with podman
-       if imagePullPolicy == always || (imagePullPolicy == missing && !containerImageExists(imgref, cntArch)) {
+       if imagePullPolicy == pullPolicyAlways || (imagePullPolicy == pullPolicyMissing && !containerImageExists(imgref, cntArch)) {
                logrus.Infof("Pulling image %s (arch=%s)\n", imgref, cntArch)
                cmd := exec.Command(
                        "podman",
@@ -599,7 +611,7 @@ func run() error {
        manifestCmd.Flags().StringArray("type", []string{"qcow2"}, fmt.Sprintf("image types to build [%s]", allImageTypesString()))
        manifestCmd.Flags().Bool("local", false, "use a local container rather than a container from a registry")
        manifestCmd.Flags().String("rootfs", "", "Root filesystem type. If not given, the default configured in the source container image is used.")
-       manifestCmd.Flags().String("pull", string(missing), "pull policy ['always', 'never', 'missing]")
+       manifestCmd.Flags().String("pull", string(pullPolicyMissing), "pull policy ['always', 'never', 'missing]")
        _ = manifestCmd.Flags().MarkDeprecated("local", "use pull=never instead")
 
        buildCmd.Flags().AddFlagSet(manifestCmd.Flags())
```

Reply:
Thanks, yeah I like this. I'll update the PR :)